From 981f27b897d68b3340b3485474aec17f2f54cd4a Mon Sep 17 00:00:00 2001
From: Caio Marcelo de Oliveira Filho <caio.oliveira@intel.com>
Date: Tue, 9 Oct 2018 13:43:37 -0700
Subject: [PATCH 1/2] Do not force downloading ExternalProjects

OFFLINE_BUILDS is used to avoid update being called, but still clones
the dependencies from git the first time.  Change so it looks at a
directory instead.

TODO: Add a build option for toggling this behavior so can be upstreamed.
---
 CMakeLists.txt       | 3 +--
 tests/CMakeLists.txt | 4 +---
 2 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index eae850a..be2fe4c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,10 +25,9 @@ enable_testing()
 if (OFFLINE_BUILDS)
   include(ExternalProject)
   ExternalProject_Add(bcc
-    GIT_REPOSITORY https://github.com/iovisor/bcc
+    SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/bcc
     STEP_TARGETS build update
     EXCLUDE_FROM_ALL 1
-    UPDATE_DISCONNECTED 1
     BUILD_COMMAND ${CMAKE_COMMAND} --build . --target bcc-static
     )
 else()
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index c283efa..799b2cf 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -39,11 +39,9 @@ find_package(Threads REQUIRED)
 if (OFFLINE_BUILDS)
   include(ExternalProject)
   ExternalProject_Add(gtest-git
-    GIT_REPOSITORY https://github.com/google/googletest.git
-    GIT_TAG release-1.8.1
+    SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../3rdparty/gtest
     STEP_TARGETS build update
     EXCLUDE_FROM_ALL 1
-    UPDATE_DISCONNECTED 1
     )
 else()
   include(ExternalProject)
-- 
2.19.1

